polaris-core/src/main/java/org/apache/polaris/core/PolarisConfiguration.java

```diff
@@ -264,4 +264,13 @@ public static <T> Builder<T> builder() {
                   + STORAGE_CREDENTIAL_DURATION_SECONDS.key)
           .defaultValue(30 * 60) // 30 minutes
           .build();
+
+  // The default value is kept at 12 for backward compatibility
+  public static final PolarisConfiguration<Integer> MAX_METADATA_REFRESH_RETRIES =
+      PolarisConfiguration.<Integer>builder()
+          .key("MAX_METADATA_REFRESH_RETRIES")
+          .description(
+              "How many times to retry refreshing metadata when the previous error was retryable")
+          .defaultValue(12)
+          .build();
```

Comment:
I agree with the backward compatibility,
[doubt] These retries are on top of SDK retries ? 
why are we did we go with a such high number ? default for hive catalog impl is 2 more details 
: https://github.com/apache/iceberg/pull/5510#issuecomment-1226712167

Reply:
> These retries are on top of SDK retries

I believe so. This is done by Iceberg `Task` under the hood. Is it possible to make the default is reasonable number while changing the current deployment config? The concern is that users normally don't change the default value until they hit issues.

Reply:
> current deployment config

Curious what you mean by this. Do you mean setting this in one of the k8s/application properties files?

I'd be very much in favor of lowering the default as 12 is quite large and potentially dangerous. I'd be surprised if anyone is getting benefit from it being so high.

Reply:
> Do you mean setting this in one of the k8s/application properties files?

I was talking about the Snowfalke's Polaris deployment.

> I'd be surprised if anyone is getting benefit from it being so high.

I believe not. The only compatibility concern is the current production deployment, including Snowflake, Zsaler, etc. Given that we don't have a release yet, the impact will manageable. In general, it should be fine to prefer a reasonable retry number(2 or 3) to compatibility. cc @MonkeyCanCode

Reply:
Updated it to 2!